### PR TITLE
Remove non-existent property

### PR DESCRIPTION
**Pull request by inkydragon**
@staticfloat

**tl; dr**

1. rm if branch in `Base.getproperty(::SHA2_CTX, )`
2. Make function `getproperty(::SHA2_CTX, )` and function `getproperty(::SHA3_CTX, )` error messages consistent
3. rm trailing spaces in test
4. add more test for `Base.getproperty`

### 1. rm if branch

Property `W` only exists in `SHA1_CTX`, not exists in `<:SHA2_CTX`.

https://github.com/JuliaCrypto/SHA.jl/blob/918aff1fd132034d19a940894106db2c97d8e977/src/types.jl#L10-L15

https://github.com/JuliaCrypto/SHA.jl/blob/918aff1fd132034d19a940894106db2c97d8e977/src/types.jl#L18-L22

### 2. error msg

https://github.com/JuliaCrypto/SHA.jl/blob/918aff1fd132034d19a940894106db2c97d8e977/src/types.jl#L52

https://github.com/JuliaCrypto/SHA.jl/blob/918aff1fd132034d19a940894106db2c97d8e977/src/types.jl#L100

### 3. more test

https://app.codecov.io/gh/JuliaCrypto/SHA.jl/blob/master/src/types.jl

----

And furthermore, maybe we can delete the entire `Base.getproperty(::Union{SHA2_CTX, SHA3_CTX}, ...)`.

### CODE REVIEW

**Comment by staticfloat** [comment]
This appears to have slipped through the cracks, I'm sorry.  Can you resolve the conflicts?

**Comment by staticfloat** [comment]
Thanks for taking that over the finish line, @inkydragon !
